packages/@react-stately/menu/src/useMenuTriggerState.ts

```diff
@@ -23,16 +23,52 @@ export interface MenuTriggerState extends OverlayTriggerState {
   open(focusStrategy?: FocusStrategy | null): void,
 
   /** Toggles the menu. */
-  toggle(focusStrategy?: FocusStrategy | null): void
+  toggle(focusStrategy?: FocusStrategy | null): void,
+
+  /** Closes the menu and all submenus in the menu tree. */
+  close: () => void,
+
+  /** Opens a specific submenu tied to a specific menu item at a specific level. */
+  UNSTABLE_openSubmenu: (triggerKey: Key, level: number) => void,
+
+  /** Closes a specific submenu tied to a specific menu item at a specific level. */
+  UNSTABLE_closeSubmenu: (triggerKey: Key, level: number) => void,
+
+  /** An array of open submenu trigger keys within the menu tree.
+   * The index of key within array matches the submenu level in the tree.
+   */
+  UNSTABLE_expandedKeysStack: Key[]
 }
 
 /**
  * Manages state for a menu trigger. Tracks whether the menu is currently open,
- * and controls which item will receive focus when it opens.
+ * and controls which item will receive focus when it opens. Also tracks the open submenus within
+ * the menu tree via their trigger keys.
  */
 export function useMenuTriggerState(props: MenuTriggerProps): MenuTriggerState  {
   let overlayTriggerState = useOverlayTriggerState(props);
   let [focusStrategy, setFocusStrategy] = useState<FocusStrategy>(null);
+  let [expandedKeysStack, setExpandedKeysStack] = useState<string[]>([]);
+  let closeAll = () => {
+    setExpandedKeysStack([]);
+    overlayTriggerState.close();
+  };
+
+  // TODO: if level > the length of the expandedKeyStack then this doesn't quite work.
```

Comment:
Theoretically it should never happen since it should never skip levels when opening a submenu (e.g. if I have 3 levels of submenus open, I can't then call "open level 7's sub menu"). I'll add a check here that guards against that case.